src/modeling_library/distributions/binom.jl

```diff
@@ -0,0 +1,28 @@
+struct Binomial <: Distribution{Int} end
+
+"""
+	binom(n::Integer, p::Real)
+Sample an `Int` from the Binomial distribution with parameters `n` and `p`.
+"""
+const binom = Binomial()
+
+function logpdf(::Binomial, x::Integer, n::Integer, p::Real)
+	if x < 0 return -Inf end
+	coefficient = loggamma(n + 1) - loggamma(n + 1 - x) - loggamma(x + 1)
+	coefficient + x * log(p) + (n - x) * log(1 - p)
+end
+
+function logpdf_grad(::Binomial, x::Integer, n::Integer, p::Real)
+	(nothing, nothing, (x / p - (n - x) / (1 - p)))
+end
+
+function random(::Binomial, n::Integer, p::Real)
+	rand(Distributions.Binomial(n, p))
```

Comment:
It's how you sample from a `Distributions.jl` distribution! Almost all the Gen distributions are implemented that way, see for e.g.:

https://github.com/probcomp/Gen/blob/8ca4dc24844f93502b0f0a766f35673cd82a1ede/src/modeling_library/geometric.jl

Reply:
Oh, looks like Distributions.jl is doing something funky (but reasonable) -- since AFAIK the spec of `Base.rand(::Type{T})` is to return a random instance of `T`, and is not geared toward allowing more than one possible distribution on the same type of value, they are bending that rule a little (or crazily subclassing `Integer`).  :+1: